shared/src/commonMain/kotlin/dev/johnoreilly/confetti/ApolloClientCache.kt

```diff
@@ -0,0 +1,42 @@
+package dev.johnoreilly.confetti
+
+import com.apollographql.apollo3.ApolloClient
+import com.apollographql.apollo3.cache.normalized.api.MemoryCacheFactory
+import com.apollographql.apollo3.cache.normalized.normalizedCache
+import com.apollographql.apollo3.cache.normalized.sql.SqlNormalizedCacheFactory
+import dev.johnoreilly.confetti.di.getDatabaseName
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.get
+
+class ApolloClientCache : KoinComponent {
```

Comment:
does this need to be a `KoinComponent` ?

Reply:
There is a call below to 

```
return get<ApolloClient.Builder>()
```

Reply:
Suggestions?